### PR TITLE
Fix #979:Enhancing collection sheet

### LINKED ISSUE #979: Enhancing Individual Collection Sheet
**Summary:** 

Enhancing the Overall UI and the user experience of the Individual collection sheet.

**Screenshots:** 

screenshot_2018-05-18-16-02-01

### PR DESCRIPTION
**Pull request by AkshGautam**
Fix #979 
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/collectionsheetindividualdetails/IndividualCollectionSheetDetailsFragment.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+    @Override
+    public void showSuccess() {
+        Toast.makeText(getContext(), getStringMessage(R.string.collectionsheet_submit_success),
+                Toast.LENGTH_LONG).show();
+    }
+
+    @Override
+    public void showError(String error) {
+
```
[the remark this answers is not in the document]
Comment (AkshGautam) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/collectionsheetindividualdetails/IndividualCollectionSheetDetailsFragment.java]: Adding an error toast.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/collectionsheetpaymentdetail/PaymentDetailsFragment.java [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+import java.util.Locale;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+/**
+ * Created by aksh on 21/6/18.
+ */
+
+public class PaymentDetailsFragment extends MifosBaseFragment
```
[the remark this answers is not in the document]
Comment (AkshGautam) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/collectionsheetpaymentdetail/PaymentDetailsFragment.java]: There was nothing I felt which could be put in presenter logic, such as a query to model etc.  If you feel there is a need of stub presenter and interface, I can add that for later usage. What's your take on this?

[the remark this answers is not in the document]
Comment (AkshGautam) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/collectionsheetpaymentdetail/PaymentDetailsFragment.java]: @droidchef done

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/views/SquareImageView.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.widget.ImageView;
+
+/**
+ * Created by aksh on 27/6/18.
+ */
+
+public class SquareImageView extends ImageView {
```
[the remark this answers is not in the document]
Comment (AkshGautam) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/views/SquareImageView.java]: Done, that was silly. :+1:

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/collectionsheetindividualdetails/IndividualCollectionSheetDetailsFragment.java [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+        sheetsAdapter.setSheetItemClickListener(this);
+        sheetsAdapter.setLoans(loansAndClientNames);
+        sheetsAdapter.setPaymentTypeList(paymentTypeList);
+        sheetsAdapter.setPaymentTypeOptionsList(sheet.getPaymentTypeOptions());
+        sheetsAdapter.notifyDataSetChanged();
+    }
+
+    @Override
+    public void showSuccess() {
+        Toast.makeText(getContext(), getStringMessage(R.string.collectionsheet_submit_success),
```
Comment (tarun0) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/collectionsheetindividualdetails/IndividualCollectionSheetDetailsFragment.java]: You are using `Toaster` everywhere in this class except for this message. For the sake of consistency, use the same class here as well.

**Comment by ethan627hsu** [comment]
Hey @AkshGautam, I see that your Travis CI build has failed, but after looking through the stack trace, I have a solution to your problem 😄!

Your error seems to be that you have a PMD rule violation. PMD is a code analyzer that helps keep code consistent with patterns, like a linter. So in basic, you have a code format error, where gradle doesn't like the way your code is written. 

**To fix the problem**, you can run PMD locally. You can achieve this by running the gradle task on mac terminal:
`./gradlew build mifosng-android:pmd --stacktrace`
or on windows command prompt:
`gradlew.bat build mifosng-android:pmd --stacktrace`

When running it, you should see the log give you a file path, where you can view the PMD report in HTML. Copy the file path with the prefix and put it into a web browser. When opening the PMD report, you should be able to find the file that has the error, and then the error!

**Pull request merged**